src/NuGet.Core/NuGet.Common/CryptoHashProvider.cs

```diff
@@ -50,29 +47,18 @@ public CryptoHashProvider(string hashAlgorithm)
                      !hashAlgorithm.Equals(SHA256HashAlgorithm, StringComparison.OrdinalIgnoreCase))
             {
                 // Only support a vetted list of hash algorithms.
-                throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, Strings.UnsupportedHashAlgorithm, hashAlgorithm), "hashAlgorithm");
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.UnsupportedHashAlgorithm, hashAlgorithm), "hashAlgorithm");
             }
 
             _hashAlgorithm = hashAlgorithm;
         }
 
         /// <summary>
-        /// Determines if we are to only allow Fips compliant algorithms.
-        /// </summary>
-        /// <remarks>
-        /// CryptoConfig.AllowOnlyFipsAlgorithm does not exist in Mono.
-        /// </remarks>
-        private static bool AllowOnlyFipsAlgorithms
-        {
-            get { return ReadFipsConfigValue(); }
-        }
-
-        /// <summary>
         /// Calculates the hash for a given stream
         /// </summary>
         public byte[] CalculateHash(Stream stream)
         {
-            using (var hashAlgorithm = GetHashAlgorithm())
+            using (var hashAlgorithm = CryptoHashUtility.GetHashAlgorithm(_hashAlgorithm))
```

Comment:
>GetHashAlgorithm [](start = 57, length = 16)

Is there a reason why the `hashAlgorithm` isn't cached? `GetHashAlgorithm` uses reflection to fetch the `hashAlgorithm`, which is incredibly expensive if `CalculateHash` gets called for every entry in the nupkg.

Reply:
No reason, looks like it should be cached